src/line_cursor.ts

```diff
@@ -312,11 +312,13 @@ export class LineCursor extends Marker {
    * @param node The current position in the AST.
    * @param isValid A function true/false depending on whether the given node
    *     should be traversed.
+   * @param loop A boolean for whether to try to loop back to the top block.
    * @returns The next node in the traversal.
    */
   getNextNode(
     node: ASTNode | null,
     isValid: (p1: ASTNode | null) => boolean,
+    loop = true,
```

Comment:
Please also add a loop param to `getPrevNode`

Reply:
FYI Grace won't be around until tomorrow so please feel free to take this and run with it.

We may need should pass loop to the recursive calls too.